File: task_1/solution.js
const readline = require("readline");
const rl = readline.createInterface(process.stdin, process.stdout);

const checkConversion = (from, to) => {
    if (from === to) {
        // Если подстроки уже равны
        return 1;
    }

    const fromSet = new Set(from.split(''));
    const toSet = new Set(to.split(''));

    if (from.length !== to.length || (fromSet.size === 33 && toSet.size === 33)) {
        // Если длина подстрок не равна
        // Или количество уникальных букв в обеих подстроках равно 33
        return 0;
    }

    const symbolsMap = {};
    for (let i = 0; i < from.length; i++) {
        if (symbolsMap[from[i]] && symbolsMap[from[i]] !== to[i]) {
            // Если мы пытаемся заменить одну букву на две разных
            return 0;
        }

        symbolsMap[from[i]] = to[i]
    }

    return 1
}

rl.on('line', (line) => {
    const [ from, to ] = line.split(' ');
    console.log(String(checkConversion(from, to)))
    rl.close();
}).on('close', () => process.exit(0));

File: task_1/solution.py
def check_conversion(str_from, str_to):
    if str_from == str_to:
        # Если подстроки уже равны
        return 1

    if len(str_from) != len(str_to) or len(set(str_from)) == len(set(str_to)) == 33:
        # Если длина подстрок не равна
        # Или количество уникальных букв в обеих подстроках равно 33
        return 0

    symbols_map = {}
    for symbol_from, symbol_to in zip(str_from, str_to):
        if symbols_map.get(symbol_from, symbol_to) != symbol_to:
            # Если мы пытаемся заменить одну букву на две разных
            return 0

        symbols_map.update({ symbol_from: symbol_to })

    return 1

str_from, str_to = input().split()
print(check_conversion(str_from, str_to))
